public/app/features/plugins/admin/api.ts

```diff
@@ -26,6 +26,7 @@ export async function getPluginDetails(id: string): Promise<CatalogPluginDetails
     readme: localReadme || remote?.readme,
     versions,
     statusContext: remote?.statusContext ?? '',
+    iam: remote?.json?.iam ?? undefined,
```

Comment:
nit: doesn't [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) mean this will result in `undefined` if remote or json aren't defined? Not sure what the need is for the nullish coalesce. 🤔

Reply:
If both results are undefined, that means the plug-on doesn't require any permissions. Do you suggest an empty object instead?

Reply:
What I mean is `remote?.json?.iam` should return `undefined` unless the properties exist. Which is why I'm wondering if we need the `?? undefined` part. There's no harm in it I'm just being curious.

![image](https://github.com/grafana/grafana/assets/73201/ced00ece-f8bf-4e21-b64f-dccea08736dd)

>The optional chaining (?.) operator accesses an object's property or calls a function. If the object accessed or function called using this operator is [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) or [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null), the expression short circuits and evaluates to [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) instead of throwing an error.

Reply:
You are right. If either values of the optional chaining are not defined, the end result will be undefined. I have made the changes as suggested.